starport/pkg/cosmosgen/generate.go

```diff
@@ -63,7 +63,7 @@ func (g *generator) setup() (err error) {
 		if err != nil {
 			return err
 		}
-		modules, err := g.discoverModules(path)
+		modules, err := g.discoverModules(path, "")
```

Comment:
we should also pass it for dependencies as they can contain invalid proto files for testing or other purposes

Reply:
But each dependency can have a proto file in a different place, right? How can we handle it?

Reply:
that's right. this can be also configured in the config.yml per dependency basis with a default `proto/`. but this might be too much for now. how about if we pass some known patterns to ignore [here](https://github.com/tendermint/starport/blob/d8dff73a47647e53650030245beb6dc8f6569753/starport/pkg/cosmosanalysis/module/module.go#L210)? Starting from the ethermint example, we can make it ignore `**/test/**` and `**/tests/**`.

Reply:
I like the pattern idea. It seems awesome. If we move the `build.proto.path` to a list, we can also define paths to skip, like `["proto/**, !proto/node_modules"]`

Let's make this PR simple for a quick fix for the `v0.17.1` and create an issue in the repo. What do you think?